pkg/segment/writer/segstore.go

```diff
@@ -71,7 +71,7 @@ type SegStore struct {
 	lastUpdated        time.Time
 	VirtualTableName   string
 	RecordCount        int
-	AllSeenColumns     map[string]bool
+	AllSeenColumns     map[string]int8 // Map of Column to Column Value size. The value is a positive int if the size is consistent across records and -1 if it is not.
```

Comment:
since we are repurposing this var for multiple things, maybe we should rename this to something like `AllSeenColumnsAndSizes` (you could find a better name than my suggestion)